actors/builtin/power/power_actor.go

```diff
@@ -472,15 +471,39 @@ func (a Actor) processDeferredCronEvents(rt Runtime) error {
 	})
 
 	for _, event := range cronEvents {
-		_, _ = rt.Send(
+		_, code := rt.Send(
 			event.MinerAddr,
 			builtin.MethodsMiner.OnDeferredCronEvent,
 			vmr.CBORBytes(event.CallbackPayload),
 			abi.NewTokenAmount(0),
 		)
-		// The exit code is ignored. If a callback fails, this actor continues to invoke other callbacks
+		// If a callback fails, this actor continues to invoke other callbacks
 		// and persists state removing the failed event from the event queue. It won't be tried again.
-		// A log message would really help here, though.
+		// Failures are unexpected here but will result in removal of miner power
+		// A log message would really help here.
+		if code != exitcode.Ok {
+			rt.State().Transaction(&st, func() interface{} {
+				store := adt.AsStore(rt)
+				// Remove power and leave miner frozen
+				claim, found, err := st.GetClaim(store, event.MinerAddr)
+				if err != nil {
+					// TODO #564 log this, failing without deducting power: bad
+					return nil
+				}
+				if !found {
+					// TODO #564 log this, failing without deducting power, but power doesn't exist so maybe ok?
+					return nil
```

Comment:
I can totally see this happening if a miner is deleted. Yes we should log but it's probably fine (the exit code should indicate this).